### PR TITLE
return empty list if sender is not found in pool for sender

### PR DESCRIPTION
**Pull request by bogdan-rosianu**
in case no tx is found in the pool for a sender, return an empty slice instead of an error.
not applicable for `last nonce` because `0` would be confusing (`0` vs `null`)

### CODE REVIEW

File: node/external/transactionAPI/apiTransactionProcessor_test.go [hunk 1]
```diff
@@ -794,6 +794,14 @@ func TestApiTransactionProcessor_GetTransactionsPoolForSender(t *testing.T) {
 		require.Equal(t, expectedHashes[i], tx.TxFields[hashField])
 		require.Equal(t, sender, tx.TxFields["sender"])
 	}
+
+	// if no tx if found in pool for a sender, it isn't an error, but return empty slice
```
Comment (ssd04) [inline node/external/transactionAPI/apiTransactionProcessor_test.go:798]:
```suggestion
	// if no tx is found in pool for a sender, it isn't an error, but return empty slice
```

**Review by iulianpascalau — COMMENTED** [review]
only what @ssd04 found.

File: node/external/transactionAPI/apiTransactionProcessor_test.go [hunk 2]
```diff
@@ -951,6 +959,15 @@ func TestApiTransactionProcessor_GetTransactionsPoolNonceGapsForSender(t *testin
 	res, err := atp.GetTransactionsPoolNonceGapsForSender(sender)
 	require.NoError(t, err)
 	require.Equal(t, expectedResponse, res)
+
+	// if no tx if found in pool for a sender, it isn't an error, but return empty slice
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply node/external/transactionAPI/apiTransactionProcessor_test.go]: done

File: node/external/transactionAPI/apiTransactionProcessor_test.go (hunk 1 above)
Comment (bogdan-rosianu) [reply node/external/transactionAPI/apiTransactionProcessor_test.go]: done

**Review by iulianpascalau — APPROVED** [review]

**Review by ssd04 — APPROVED** [review]

**Review by popenta — APPROVED** [review]
@@                            Log scanner                                  @@
##                            txs-pool-api-updates                         ##
================================================================================
- Known Warnings              27
- New Warnings                11
- Known Errors                0
- New Errors                  2
- Panics                      0
================================================================================

system test + scenario passed

**Pull request merged**